### PR TITLE
[DT-529] Add support for deferred install tasks and add long extensions

### PR DESCRIPTION
**Pull request by pedroagostinhoaptoide**
**What does this PR do?**

   - Adds support to deferred app installations, by adding the necessary logic in the install manager and App and Task implementations.
   - Adds long primitive type extension functions related to dealing with app sizes.

**Database changed?**

   No

**Where should the reviewer start?**

- See by commit.

**How should this be manually tested?**

  Flow on how to test this or QA Tickets related to this use-case: [DT-529](https://aptoide.atlassian.net/browse/DT-529)

**What are the relevant tickets?**

  Tickets related to this pull-request: [DT-529](https://aptoide.atlassian.net/browse/DT-529)

**Code Review Checklist**

- [ ] Documentation on public interfaces
- [ ] Database changed? If yes - Migration?
- [ ] Remove comments & unused code & forgotten testing Logs
- [ ] Codestyle
- [ ] New Kotlin code has unit tests
- [ ] New flows in presenters unit tests
- [ ] Mappers/Validators with any kind of logic unit tests
- [ ] Functional tests pass

[DT-529]: https://aptoide.atlassian.net/browse/DT-529?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DT-529]: https://aptoide.atlassian.net/browse/DT-529?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: install-manager/src/main/java/cm/aptoide/pt/install_manager/App.kt [hunk 1]
```diff
@@ -27,7 +27,11 @@ interface App {
    * @throws IllegalStateException if another task is already running
    * @throws IllegalArgumentException if same or newer version is already known to be installed
    */
-  suspend fun install(installPackageInfo: InstallPackageInfo): Task
+  suspend fun install(
```
Comment (ricardo-frango) [inline install-manager/src/main/java/cm/aptoide/pt/install_manager/App.kt:30]: missing new params in java doc

File: install-manager/src/main/java/cm/aptoide/pt/install_manager/RealTask.kt [hunk 2]
```diff
@@ -19,6 +19,7 @@ internal class RealTask internal constructor(
   private val packageDownloader: PackageDownloader,
   private val packageInstaller: PackageInstaller,
   private val taskInfoRepository: TaskInfoRepository,
+  private val deferMe: suspend (RealTask) -> Unit,
```
Comment (ricardo-frango) [inline install-manager/src/main/java/cm/aptoide/pt/install_manager/RealTask.kt:22]: unused property

File: install-manager/src/main/java/cm/aptoide/pt/install_manager/RealTask.kt [hunk 3]
```diff
@@ -85,6 +93,7 @@ internal class RealTask internal constructor(
       finalize(Task.State.ABORTED)
     } catch (e: CancellationException) {
       finalize(Task.State.CANCELED)
+    } catch (e: DeferDownloadException) {
```
Comment (ricardo-frango) [inline install-manager/src/main/java/cm/aptoide/pt/install_manager/RealTask.kt:96]: this catch is empty

File: install-manager/src/main/java/cm/aptoide/pt/install_manager/workers/PackageDownloader.kt [hunk 4]
```diff
@@ -22,7 +22,11 @@ interface PackageDownloader {
    * Flow throws anything except [CancellationException], signalling about download failure
    * Or [AbortException] if download was aborted with the reason in message
    */
-  suspend fun download(packageName: String, installPackageInfo: InstallPackageInfo): Flow<Int>
+  suspend fun download(
```
Comment (ricardo-frango) [inline install-manager/src/main/java/cm/aptoide/pt/install_manager/workers/PackageDownloader.kt:25]: missing new param in java doc

**Review by ricardo-frango — CHANGES_REQUESTED** [review]

File: install-manager/src/main/java/cm/aptoide/pt/install_manager/App.kt [hunk 5]
```diff
@@ -23,11 +23,17 @@ interface App {
    * Creates an installation task.
    *
    * @param installPackageInfo - a package info to use for the installation
+   * @param waitForWifi - if installation should wait for wifi
+   * @param onTerminate - called after installation is finished with success
```
Comment (G0retZ) [inline install-manager/src/main/java/cm/aptoide/pt/install_manager/App.kt:27]: The description is misleading

**Review by G0retZ — CHANGES_REQUESTED** [review]
Rebase on `dev-v10`

File: download-view/src/main/java/cm/aptoide/pt/download_view/presentation/DownloadView.kt
Comment (G0retZ) [inline download-view/src/main/java/cm/aptoide/pt/download_view/presentation/DownloadView.kt:1]: This change should be in a previous commit

**Review by G0retZ — APPROVED** [review]

**Review by ricardo-frango — APPROVED** [review]

**Pull request merged**